app/Extensions/WoocommerceProductLabel.php

```diff
@@ -15,18 +15,19 @@ public static function get_instance() {
 
 	public function __construct() {
 		if ( is_plugin_active( 'woocommerce-advanced-product-labels/woocommerce-advanced-product-labels.php' ) ) {
-			add_filter( 'blaze_wooless_product_data_for_typesense', array( $this, 'product_label_html' ), 10, 3 );
+			add_filter( 'blaze_wooless_product_data_for_typesense', array( $this, 'product_label_html' ), 10, 2 );
 			add_action( 'blaze_get_advance_custom_labels_html', array( \WAPL_Global_Labels::class, 'global_label_hook' ), 15 );
 		}
 	}
 
-	public function product_label_html( $product_data, $product_id, $product ) {
-
-		ob_start();
-		setup_postdata( $product->get_id() );
-		do_action( 'blaze_get_advance_custom_labels_html' );
-		$label_html                               = ob_get_clean();
-		$product_data['metaData']['productLabel'] = htmlspecialchars( $label_html, ENT_QUOTES, 'UTF-8' );
+	public function product_label_html( $product_data, $product_id ) {
+		if(!empty(product_data) && !empty($product) && $product_id) {
```

Comment:
please update. code is in wrong syntax product_data is missing $. Another thing is $product with your changes is undefined.